aamt_parse2.pm

```diff
@@ -157,6 +163,244 @@ sub pars_GetDirlistinght
+        }
+        if ($line =~ /DB_PASSWORD/)
+        {
+            $array[$sIndex][DB_PASS] = $tempValue;
+        }
+        if ($line =~ /DB_HOST/)
+        {
+            $array[$sIndex][DB_HOST] = $tempValue;
+        }
+        if ($line =~ /WP_SITEURL/)
+        {
+            $array[$sIndex][WP_SITEURL] = $tempValue;
+        }
+        if ($line =~ /INCLUDED_FILES/)
+        {
+            # @files = split /;/,$tempValue;
+            $array[$sIndex][INCLUDED_FILES] = $tempValue;
+        }
+    }
+
+}
+
+sub pars_generateTransformRule
+{
+	my $framework=$_[0];
+	my $rDatabase=$_[1];
+	my $rUsername=$_[2];
+	my $rPassword=$_[3];
+	my $rServer=$_[4];
+	my $rTargetUrl=$_[5];
+	my $configFile=&basename($_[6]);
+	    my $workingFolder = &utf_getCurrentWorkingFolder();
+    	    my $sessionname = &ilog_getSessionName();
+            my $deffile=$workingFolder.'/writephp.def';
+            open my $fh,'<',$deffile or die "can't open $deffile";
+            while( my $line = <$fh>)
+            {
+                my @tmpsplit = split('=',$line);
+		my $phpwrite = $tmpsplit[1];
+        	$phpwrite =~ s/^\s+//;
+        	$phpwrite =~ s/\s+$//;
+        	$phpwrite =~ s/[\r\n]+//;
+                my $tmpfwork=$tmpsplit[0];
+                $tmpfwork =~ s/\s+$//;
+                if($framework eq $tmpfwork)
+                {
+	                #ilog_printf(1,"php $phpwrite "."$workingFolder/$sessionname/${framework}-transform.txt $rDatabase $rUsername $rPassword $rServer $rTargetUrl");
+			`php $phpwrite "$workingFolder/$sessionname/${framework}-${configFile}-transform.txt" "$rDatabase" "$rUsername" "$rPassword" "$rServer" "$rTargetUrl" "$configFile";`;
+                        return 1;
+                }
+            }
+return 0;
+}
+
+sub pars_readTransformRule()
+{
+	my $framework=shift;
+	my $configFile=&basename(shift);
+        my $workingFolder = &utf_getCurrentWorkingFolder();
+        my $sessionname = &ilog_getSessionName();
+	my $ruleFile = "$workingFolder/$sessionname/${framework}-${configFile}-transform.txt";
+	open my $fh,'<',$ruleFile or die "can't open $ruleFile";
+	my $lineMatch = <$fh>;
+	$lineMatch =~ s/[\r\n]+//;
+	my $lineNotMatch = <$fh>;
+	$lineNotMatch =~ s/[\r\n]+//;
+	my $settingsLine ="";
+	while( my $line = <$fh>)
+	{
+		$settingsLine=$settingsLine.$line;	
+	}	
+	if($lineNotMatch)
+	{return (qr/$lineMatch/,qr/$lineNotMatch/,$settingsLine);}   
+	else
+	{return (qr/$lineMatch/,$lineNotMatch,$settingsLine);}
+}
+
+sub pars_DisplaySites
+{
+my $i;
+my $sIndex = shift;
+my $strSiteName = $array[$sIndex][SITENAME];
+my $documentRoot = $array[$sIndex][DOCUMENTROOT];
+my $framework;
+my @files;
+for ($i=0;$i <= $rowCount; $i++)
+{
+    $strSiteName = $array[$i][SITENAME];
+    $documentRoot = $array[$i][DOCUMENTROOT];
+    ($framework,@files) = &pars_DetectFramework($documentRoot);
+    {
+
+	ilog_printf(1,"[$framework]:[$strSiteName]:[$documentRoot]");
+	ilog_printf(1,"\n");
+    }
+#    ui_printline();
```

Comment:
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip printing.